### PR TITLE
[tune] Prevent deletion of checkpoint from user-initiated restores

### LINKED ISSUE #7412: "restore" deletes the restored from checkpoint
If I specify a checkpoint in the `tune.run(DQNTrainer, restore=X)` - the checkpoint `X` gets deleted!

So if there is a bug or something during the initialization, I loose my checkpoint and can no longer restart from it!

### PR DESCRIPTION
**Pull request by ujvl**
Prevent deletion of checkpoint from user-initiated restores. We can do this by just not adding it to `CheckpointManager` on trial initialization.

Also some minor documentation fixes

Closes #7412 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by richardliaw** [comment]
Can you add a test?

**Comment by ujvl** [comment]
jenkins test tune

File: python/ray/tune/analysis/experiment_analysis.py [hunk 1]
```diff
@@ -120,32 +116,30 @@ def get_all_configs(self, prefix=False):
...
-            trial(Trial): The log directory of a trial, or a trial instance.
+        Args:
+            trial (Trial): The log directory of a trial, or a trial instance.
             metric (str): key for trial info to return, e.g. "mean_accuracy".
                 "training_iteration" is used by default.
-        """
 
+        Returns:
+            A list of [path, metric] lists for all persistent checkpoints of
+            the trial.
```
Comment (richardliaw) [inline python/ray/tune/analysis/experiment_analysis.py]: the second line needs to be indented or else it breaks the docs rendering

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
jenkins retest this please

**Pull request merged**